### ISSUE TITLE
`dart format` does not put new line between 'sections'

### ISSUE THREAD

**Issue by aeb-dev** [dart-lang/dart_style#1120]
Steps to reproduce:

`dart create -t console test_format`
`cd test_format`
`dart pub add args`

Create following file under `lib`
```dart
import "dart:io";
import "package:args/args.dart";

void func() {
}
```

Call `dart format .`
As stated in [Effective Dart: Style](https://dart.dev/guides/language/effective-dart/style#ordering), the file should be changed to:
```dart
import "dart:io";

import "package:args/args.dart";

void func() {
}
```

but it does not.

* Dart SDK version: 2.17.5 (stable) (Tue Jun 21 11:05:10 2022 +0200) on "windows_x64"

**Comment by munificent** [comment]
Good catch. The formatter doesn't do any import reordering (because that can be surprisingly complex), but it could at least insert a newline between `dart:` and package imports.

**Comment by aeb-dev** [comment]
I do not know how the internals work but since it can already detected can't it be ordered easily?

![image](https://user-images.githubusercontent.com/11612804/178158410-ccd556c7-72a3-4e15-91a2-e3258a939b04.png)
![image](https://user-images.githubusercontent.com/11612804/178158445-8a978935-721a-45a0-a137-ac24e3371d1f.png)

**Comment by munificent** [comment]
Reordering code can get difficult when things like comments and commented out code is involved. To avoid potentially reordering things in ways that don't make sense, the formatter is currently restricted to only insert and remove whitespace but never change the order of any code or comments.

The warnings here come from a different tool than the formatter.

**Comment by aeb-dev** [comment]
It is from `dart analyzer`. `dart fix --apply` fixes the warning

**Comment by alexeyinkin** [comment]
Just reminding that package imports also should be separated from relative imports.
